nuxeo-runtime/nuxeo-launcher-commons/src/main/java/org/nuxeo/launcher/config/BackingServiceConfigurator.java

```diff
@@ -0,0 +1,265 @@
+        for (BackingChecker checker : getCheckers()) {
+            if (checker.acceptConfiguration(configurationGenerator)) {
+                try {
+                    Failsafe.with(retryPolicy)
+                            .onFailedAttempt(failure -> log.error(failure.getMessage())) //
+                            .onRetry((c, f,
+                                    ctx) -> log.warn(String.format("Failure %d. Retrying....", ctx.getExecutions()))) //
+                            .run(() -> checker.check(configurationGenerator)); //
+                } catch (FailsafeException e) {
+                    if (e.getCause() instanceof ConfigurationException) {
+                        throw ((ConfigurationException) e.getCause());
+                    } else {
+                        throw e;
+                    }
+                }
+            }
+        }
+    }
+
+    private RetryPolicy buildRetryPolicy() {
+        RetryPolicy retryPolicy = new RetryPolicy().withMaxRetries(0);
+
+        Properties userConfig = configurationGenerator.getUserConfig();
+        if ("true".equals((userConfig.getProperty(PARAM_RETRY_POLICY_ENABLED,"false")))) {
+
+            int maxRetries = Integer.parseInt(userConfig.getProperty(PARAM_RETRY_POLICY_MAX_RETRIES, "20"));
+            int delay = Integer.parseInt(userConfig.getProperty(PARAM_RETRY_POLICY_DELAY_IN_MS, "5000"));
+
+            retryPolicy = retryPolicy.retryOn(ConfigurationException.class).withMaxRetries(maxRetries).withDelay(delay,
+                    TimeUnit.MILLISECONDS);
+        }
+        return retryPolicy;
+    }
+
+    protected Collection<BackingChecker> getCheckers() throws ConfigurationException {
+
+        if (checkers == null) {
+            checkers = new HashSet<>();
+
+            for (String template : configurationGenerator.getTemplateList()) {
+                try {
+                    File templateDir = configurationGenerator.getTemplateConf(template).getParentFile();
+                    String classPath = getClasspathForTemplate(template);
+                    String checkClass = configurationGenerator.getUserConfig().getProperty(template + ".check.class");
+
+                    Optional<URLClassLoader> ucl = getClassLoaderForTemplate(templateDir, classPath);
+                    if (ucl.isPresent()) {
+                        Class<?> klass = Class.forName(checkClass, true, ucl.get());
+                        checkers.add((BackingChecker) klass.newInstance());
+                    }
+
+                } catch (IOException e) {
+                    log.warn("Unable to read check configuration for template : " + template, e);
+                } catch (InstantiationException | IllegalAccessException | ClassNotFoundException e) {
+                    throw new ConfigurationException("Unable to check configuration for backing service " + template,
+                            e);
+                }
+            }
+            checkers.add(new DBCheck());
```

Comment:
Can't `DBCheck` be done from a configuration in a template like `common` too? At some point we'll want to be able to run with any SQL database and it would be nice to not have to change code for that.